### DISCUSSION TITLE
Simple question regarding reading Objects

### DISCUSSION THREAD

**Discussion by emanodame** [eclipse-milo/milo#1001] [General]
Following the title, the question is if it is possible to read Objects similar to how Variables can be read with Milo Library?

For example: Server Status Variable Type.

I can successfully read this variable and after constructing the json i can get this:

```
"Server Status": {
          "value": {
            "startTime": {
              "seconds": 1.647003574E9,
              "nanos": 9.6E8
            },
            "currentTime": {
              "seconds": 1.652943307E9,
              "nanos": 1.77E8
            },
            "state": {
              "value": 0.0,
              "name": "Running",
              "ordinal": 0.0
            },
            "buildInfo": {
              "productUri": "https://github.com/digitalpetri/opc-ua-demo-server",
              "manufacturerName": "digitalpetri",
              "productName": "Eclipse Milo OPC UA Demo Server",
              "softwareVersion": "0.3-SNAPSHOT",
              "buildNumber": "61431a94ba",
              "buildDate": {
                "seconds": 1.627150708E9,
                "nanos": 0.0
              }
            },
            "secondsTillShutdown": 0.0
          },
```

However when I try and read Server Object: 

```
val client = new UaClient()
val result = client.read(0.0, TimestampsToReturn.Both, Arrays.asList(Identifiers.Server), Arrays.asList(uint(0))).get()
``` 

I receive the error:

```
DataValue{value=Variant{value=null}, status=StatusCode{name=Bad_AttributeIdInvalid, value=0x80350000, quality=bad}, sourceTime=DateTime{utcTime=0, javaDate=Mon Jan 01 01:00:00 CET 1601}, serverTime=DateTime{utcTime=0, javaDate=Mon Jan 01 01:00:00 CET 1601}}
```
Bad_AttributeIdInvalid.

Any thoughts on this?

For example: Server `ns=0, i=2253`.

**Comment by kevinherron** [comment]
No, this isn't possible, because Objects do not have a Value attribute.

**Comment by emanodame** [comment]
Thanks for the prompt response. Just to add, is this not possible with Milo library or is this not generally possible with OPC-UA itself?
In other words, does OPC-UA expose ways to read Objects?

**Reply by kevinherron** [reply]
Not possible in OPC UA.

You can read any Object attribute you want, but Value is not one of them.

I think your confusion stems from the fact that the ServerStatus Node is a complex Variable that both mirrors its Value's structure in the address space as their own Nodes _and_ has a structured Value representing the same contents.

Objects do not have Values, they only aggregate/organize other Objects and Variables, there is nothing to read that represents all of the Nodes underneath.

**Reply by emanodame** [reply]
Got it. Thank you Kevin.
